### DISCUSSION TITLE
Nextjs Image Cache Invalidation

### DISCUSSION THREAD

**Discussion by Louai99k** [vercel/next.js#35276] [Help]
I'm using aws s3 buckets to store my assets. When any of the existed assets is changed from the front-end (eg: client changed his profile image) the backend I made will change the asset only without touching the url. Now here is my problem in my website I'm using **revalidate** in **getStaticProps** but it's not helping because the link is actually the same and there is nothing to update but the cach. So is there any way to update those images caches programmatically? (as I mentioned when user update his profile image or something like that).

By the way when I click on DevTools and see the preview from the Network tab it shows the right version of the asset

The problem I mentioned is very similar to [this](https://support.cloudinary.com/hc/en-us/articles/202520852-I-ve-replaced-an-existing-image-with-a-new-one-but-my-website-still-shows-the-old-one-Why-is-that-) if there is any thing not clear

**Comment by hmartiins** [comment]
It's been a few months since your question, did you manage to implement any solution?

**Reply by edgarasben** [reply]
I started adding a timestamp to updated images as such, so a browser knows it's a new image. E.g.:
`https://example.com/images/avatars/2342db78-5fc1-4dd2-b201-50b12b04b3f4.jpg?1662466142286`

It solved the issue.

**Reply by hmartiins** [reply]
Got it, I ended up finding this solution on the web, I think I'll have to follow this path
thnks

**Reply by iduuck** [reply]
While this is pretty late, I would not suggest adding a timestamp, because then you don't need the `next/image` tag, since the image will be optimized on each render, and you could just add the `unoptimized` flag to the component.
